mina-sshd-api-core/src/main/test/io/jenkins/plugins/mina_sshd_api/core/authenticators/MinaSSHPasswordKeyAuthenticatorTest.java

```diff
@@ -0,0 +1,111 @@
+package io.jenkins.plugins.mina_sshd_api.core.authenticators;
+
+import com.cloudbees.jenkins.plugins.sshcredentials.SSHAuthenticator;
+import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl;
+import hudson.model.TaskListener;
+import org.apache.sshd.client.SshClient;
+import org.apache.sshd.client.session.ClientSession;
+import org.apache.sshd.server.SshServer;
+import org.apache.sshd.server.auth.password.UserAuthPasswordFactory;
+import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class MinaSSHPasswordKeyAuthenticatorTest {
+
+    private static final Logger LOGGER = Logger.getLogger(MinaSSHPasswordKeyAuthenticatorTest.class.getName());
+
+    private SshServer sshd;
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    private final StandardUsernamePasswordCredentials user =
+        new UsernamePasswordCredentialsImpl(CredentialsScope.SYSTEM, null, "foobar", "foobar", "foomanchu");
+
+    @After
+    public void tearDown() {
+        if (sshd != null) {
+            try {
+                sshd.stop(true);
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "Problems shutting down ssh server", e);
+            }
+        }
+    }
+
+    @Before
+    public void setUp() throws IOException {
+        sshd = SshServer.setUpDefaultServer();
+        sshd.setHost("localhost");
+        sshd.setPort(0);
+        sshd.setKeyPairProvider(new SimpleGeneratorHostKeyProvider());
+        sshd.setPasswordAuthenticator((s, s1, serverSession) -> user.getUsername().equals(s) && user.getPassword().getPlainText().equals(s1));
+        sshd.setUserAuthFactories(Collections.singletonList(new UserAuthPasswordFactory()));
+        try {
+            sshd.start();
+            LOGGER.log(Level.INFO, "Started ssh Server");
+        } catch (Throwable e) {
+            LOGGER.log(Level.WARNING, "Problems starting ssh server", e);
+            try {
+                sshd.stop();
+            } catch (Throwable t) {
+                LOGGER.log(Level.WARNING, "Problems shutting down ssh server", t);
+            }
+            throw e;
+        }
+    }
+
+    @Test
+    public void testAuthenticate() throws Exception {
```

Comment:
I think this can just be deleted? `testFactory` already verifies that the public API works as intended.